tests/lib/thrift/ThriftTest/CMakeLists.txt

```diff
@@ -37,3 +37,6 @@ thrift(
 )
 
 target_sources(app PRIVATE ${app_sources})
+
+# needed because std::iterator was deprecated with -std=c++17
+zephyr_library_compile_options(-Wno-deprecated-declarations)
```

Comment:
This file does not have `zephyr_library()`, so not sure which library it is adding this option to.